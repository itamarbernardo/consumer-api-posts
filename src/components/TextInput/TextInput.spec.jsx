import { render, screen } from "@testing-library/react"
import {TextInput} from '.'
import userEvent from "@testing-library/user-event"


describe('<TextInput />', () => {
    it('should have a value of seachValue', () => {
        const fn = jest.fn()
        render(<TextInput handleChange={fn} searchValue="Testando"/>)

        const input = screen.getByPlaceholderText(/Digite sua busca/i)
        expect(input).toBeInTheDocument()

        expect(input.value).toBe('Testando')
    })

    it('should call handleChange function on each key pressed', async () => {
        const fn = jest.fn()
        render(<TextInput handleChange={fn} />)
        
        const input = screen.getByPlaceholderText(/Digite sua busca/i)
        const value = 'o valor'

        await userEvent.type(input, value) //Digita esse valor no input
        expect(input.value).toBe(value)
        expect(fn).toHaveBeenCalledTimes(value.length) //chamou essa funcao 7 vezes pois value tem 7 caracteres
    })


    it('shoud match snapshot', () => {
        const fn = jest.fn()
        const {container} = render(<TextInput handleChange={fn} searchValue="Testando"/>)

        expect(container.firstChild).toMatchSnapshot() //Cria uma pasta com o snapshot pra comparar com o teste posteriormente
    })
})