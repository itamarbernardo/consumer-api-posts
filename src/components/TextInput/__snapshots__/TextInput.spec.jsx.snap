// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> shoud match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Digite sua busca..."
  type="search"
  value="Testando"
/>
`;
